server/src/internalClusterTest/java/org/opensearch/index/codec/CodecIT.java

```diff
@@ -0,0 +1,167 @@
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class CodecIT extends OpenSearchIntegTestCase {
+
+    public void testForceMergeMultipleCodecs() throws ExecutionException, InterruptedException {
+
+        Map<String, String> codecMap = Map.of(
+            "best_compression",
+            "BEST_COMPRESSION",
+            "zstd_no_dict",
+            "ZSTD_NO_DICT",
+            "zstd",
+            "ZSTD",
+            "default",
+            "BEST_SPEED"
+        );
+
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            forceMergeMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
+        }
+
+    }
+
+    private void forceMergeMultipleCodecs(String finalCodec, String finalCodecMode, Map<String, String> codecMap) throws ExecutionException,
+        InterruptedException {
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String index = "test-index" + finalCodec;
+
+        // creating index
+        createIndex(
+            index,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put("index.codec", "default")
+                .build()
+        );
+        ensureGreen(index);
+
+        // ingesting and asserting segment codec mode for all four codecs
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            assertSegmentCodec(index, codec.getKey(), codec.getValue());
+        }
+
+        // force merge into final codec
+        assertSegmentCodec(index, finalCodec, finalCodecMode);
+        final ForceMergeResponse forceMergeResponse = client().admin().indices().prepareForceMerge(index).setMaxNumSegments(1).get();
+
+        assertThat(forceMergeResponse.getFailedShards(), is(0));
+        assertThat(forceMergeResponse.getSuccessfulShards(), is(2));
+
+        flushAndRefreshIndex(index);
+
+        List<Segment> segments = getSegments(index).stream().filter(Segment::isSearch).collect(Collectors.toList());
+        assertEquals(1, segments.size());
+        assertTrue(segments.stream().findFirst().get().attributes.containsValue(finalCodecMode));
+    }
+
+    private void assertSegmentCodec(String index, String codec, String codecMode) throws InterruptedException, ExecutionException {
+
+        assertAcked(client().admin().indices().prepareClose(index));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index));
+
+        ingest(index);
+        flushAndRefreshIndex(index);
+
+        assertTrue(getSegments(index).stream().anyMatch(segment -> segment.attributes.containsValue(codecMode)));
```

Comment:
Can be moved to caller and called only once to verify existence of all the codecs.

Reply:
Added in the latest commit